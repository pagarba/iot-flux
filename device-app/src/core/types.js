export default {
  LOGIN_REQUEST: 'LOGIN_REQUEST',
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_FAILURE: 'LOGIN_FAILURE',
  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  LOGOUT_FAILURE: 'LOGOUT_FAILURE',
  ACCOUNT_FETCH_SUCCESS: 'ACCOUNT_FETCH_SUCCESS',

  CREATE_USER_REQUEST: 'CREATE_USER_REQUEST',
  CREATE_USER_SUCCESS: 'CREATE_USER_SUCCESS',
  CREATE_USER_FAILURE: 'CREATE_USER_FAILURE',

  CREATE_DEVICE_REQUEST: 'CREATE_DEVICE_REQUEST',
  CREATE_DEVICE_SUCCESS: 'CREATE_DEVICE_SUCCESS',
  CREATE_DEVICE_FAILURE: 'CREATE_DEVICE_FAILURE',

  EDIT_DEVICE_REQUEST: 'EDIT_DEVICE_REQUEST',
  EDIT_DEVICE_SUCCESS: 'EDIT_DEVICE_SUCCESS',
  EDIT_DEVICE_FAILURE: 'EDIT_DEVICE_FAILURE',

  GET_DEVICES_REQUEST: 'GET_DEVICES_REQUEST',
  GET_DEVICES_SUCCESS: 'GET_DEVICES_SUCCESS',
  GET_DEVICES_FAILURE: 'GET_DEVICES_FAILURE',

  GET_DEVICE_REQUEST: 'GET_DEVICE_REQUEST',
  GET_DEVICE_SUCCESS: 'GET_DEVICE_SUCCESS',
  GET_DEVICE_FAILURE: 'GET_DEVICE_FAILURE',

  DELETE_DEVICE_REQUEST: 'DELETE_DEVICE_REQUEST',
  DELETE_DEVICE_SUCCESS: 'DELETE_DEVICE_SUCCESS',
  DELETE_DEVICE_FAILURE: 'DELETE_DEVICE_FAILURE',

  CREATE_CHANNEL_REQUEST: 'CREATE_CHANNEL_REQUEST',
  CREATE_CHANNEL_SUCCESS: 'CREATE_CHANNEL_SUCCESS',
  CREATE_CHANNEL_FAILURE: 'CREATE_CHANNEL_FAILURE',

  EDIT_CHANNEL_REQUEST: 'EDIT_CHANNEL_REQUEST',
  EDIT_CHANNEL_SUCCESS: 'EDIT_CHANNEL_SUCCESS',
  EDIT_CHANNEL_FAILURE: 'EDIT_CHANNEL_FAILURE',

  GET_CHANNELS_REQUEST: 'GET_CHANNELS_REQUEST',
  GET_CHANNELS_SUCCESS: 'GET_CHANNELS_SUCCESS',
  GET_CHANNELS_FAILURE: 'GET_CHANNELS_FAILURE',

  GET_CHANNEL_REQUEST: 'GET_CHANNEL_REQUEST',
  GET_CHANNEL_SUCCESS: 'GET_CHANNEL_SUCCESS',
  GET_CHANNEL_FAILURE: 'GET_CHANNEL_FAILURE',

  DELETE_CHANNEL_REQUEST: 'DELETE_CHANNEL_REQUEST',
  DELETE_CHANNEL_SUCCESS: 'DELETE_CHANNEL_SUCCESS',
  DELETE_CHANNEL_FAILURE: 'DELETE_CHANNEL_FAILURE',

  ADD_DEVICE_TO_CHANNEL_REQUEST: 'ADD_DEVICE_TO_CHANNEL_REQUEST',
  ADD_DEVICE_TO_CHANNEL_SUCCESS: 'ADD_DEVICE_TO_CHANNEL_SUCCESS',
  ADD_DEVICE_TO_CHANNEL_FAILURE: 'ADD_DEVICE_TO_CHANNEL_FAILURE',

  DELETE_DEVICE_FROM_CHANNEL_REQUEST: 'DELETE_DEVICE_FROM_CHANNEL_REQUEST',
  DELETE_DEVICE_FROM_CHANNEL_SUCCESS: 'DELETE_DEVICE_FROM_CHANNEL_SUCCESS',
  DELETE_DEVICE_FROM_CHANNEL_FAILURE: 'DELETE_DEVICE_FROM_CHANNEL_FAILURE',

  GET_CHANNEL_DEVICES_REQUEST: 'GET_CHANNEL_DEVICES_REQUEST',
  GET_CHANNEL_DEVICES_SUCCESS: 'GET_CHANNEL_DEVICES_SUCCESS',
  GET_CHANNEL_DEVICES_FAILURE: 'GET_CHANNEL_DEVICES_FAILURE',

  CREATE_EVENT_REQUEST: 'CREATE_EVENT_REQUEST',
  CREATE_EVENT_SUCCESS: 'CREATE_EVENT_SUCCESS',
  CREATE_EVENT_FAILURE: 'CREATE_EVENT_FAILURE',

  EDIT_EVENT_REQUEST: 'EDIT_EVENT_REQUEST',
  EDIT_EVENT_SUCCESS: 'EDIT_EVENT_SUCCESS',
  EDIT_EVENT_FAILURE: 'EDIT_EVENT_FAILURE',

  GET_EVENTS_REQUEST: 'GET_EVENTS_REQUEST',
  GET_EVENTS_SUCCESS: 'GET_EVENTS_SUCCESS',
  GET_EVENTS_FAILURE: 'GET_EVENTS_FAILURE',

  GET_EVENT_REQUEST: 'GET_EVENT_REQUEST',
  GET_EVENT_SUCCESS: 'GET_EVENT_SUCCESS',
  GET_EVENT_FAILURE: 'GET_EVENT_FAILURE',

  DELETE_EVENT_REQUEST: 'DELETE_EVENT_REQUEST',
  DELETE_EVENT_SUCCESS: 'DELETE_EVENT_SUCCESS',
  DELETE_EVENT_FAILURE: 'DELETE_EVENT_FAILURE',
}