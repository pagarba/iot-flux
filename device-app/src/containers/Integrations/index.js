import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

import Button from '@material-ui/core/Button';
import { withStyles } from "@material-ui/core/styles/index";

import JustifireLogo from '../../assets/img/justifire.png';
import VisualFlightsLogo from '../../assets/img/visual_flights.png';
import TriggerSmartLogo from '../../assets/img/trigger_smart.png';
import DataLifeLogo from '../../assets/img/datalife.png';
import GuniaryLogo from '../../assets/img/guniary.png';

import { integrations } from '../../constants';
import { newIntegrations } from '../../constants';
import qs from 'stringquery'

const logos = {
  justifire: JustifireLogo,
  visualFlights: VisualFlightsLogo,
  triggerSmart: TriggerSmartLogo,
  datalife: DataLifeLogo,
  guniary: GuniaryLogo,
}

const styles = theme => ({
  root: {
    display: 'block'
  },
  button: {
    margin: theme.spacing.unit,
    marginTop: '2em',
    color: '#fff',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexShrink: 0,
    fontWeight: 'bold',
  },
  imgContainer: {
    marginRight: '30px',
    display: 'flex',
    flexDirection: 'column',
  },
  imgList: {
    alignItems: 'center',
    display: 'flex',
    background: '#fff',
    width: '100%',
    padding: '1em',
    borderRadius: '4px',
    overflowX: 'scroll',
  },
  imgLogo: {
    height: '50px',
  },
  formControl: {
    margin: theme.spacing.unit,
  },
  myIntegrationsPanel: {
    backgroundColor: '#52cee8',
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

export class Integrations extends Component {

  constructor(props) {
    super(props);

    this.state = {
      expanded: null,
      integrations: ['justifire', 'visualFlights'],
      newIntegrations: ['triggerSmart', 'datalife', 'guniary']
    }
  }

  componentDidMount() {
    const obj = qs(this.props.location.search);
    this.setState({
      expanded: obj.mode === 'create' ? 'panel2' : 'panel1',
    });
  }

  handleChange = panel => (event, expanded) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  handleBlur = event => {
    if (!this.state[event.target.name])
      this.setState({ [event.target.name]: null });
  };

  handleInputChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  addIntegration = newIntegration => {
    let { integrations, newIntegrations } = this.state;

    integrations.push(newIntegration);
    newIntegrations = newIntegrations.filter(integration => integration !== newIntegration);

    this.setState({
      integrations,
      newIntegrations
    });
  }

  deleteIntegration = integration => {
    let { integrations, newIntegrations } = this.state;

    newIntegrations.push(integration);
    integrations = integrations.filter(e => e !== integration);

    this.setState({
      integrations,
      newIntegrations
    });
  }

  render() {
    const { classes } = this.props;
    const { expanded } = this.state;
    return (
      <div className={classes.root}>
        <ExpansionPanel
          key={1}
          className={classes.myIntegrationsPanel}
          expanded={expanded === 'panel1'}
          onChange={this.handleChange('panel1')}
        >
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon color="primary" />}>
            <Typography color="primary" className={classes.heading}>MY INTEGRATIONS</Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails>
            <div className={classes.imgList}>
              {
                this.state.integrations.map((integration, i) => (
                  <div key={i} className={classes.imgContainer}>
                    <img className={classes.imgLogo} src={logos[integration]} />
                    <Button
                      color="secondary"
                      className={classes.button}
                      onClick={() => this.deleteIntegration(integration)}
                      variant="contained"
                    >
                      + Delete {integration}
                    </Button>
                  </div>
                ))
              }
            </div>
          </ExpansionPanelDetails>
        </ExpansionPanel>
        <ExpansionPanel
          key={2}
          expanded={expanded === 'panel2'}
          onChange={this.handleChange('panel2')}
        >
          <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
            <Typography className={classes.heading}>ADD A NEW INTEGRATION</Typography>
          </ExpansionPanelSummary>
          <ExpansionPanelDetails className={classes.root}>
            <div className={classes.imgList}>
              {
                this.state.newIntegrations.map(integration => (
                  <div className={classes.imgContainer}>
                    <img className={classes.imgLogo} src={logos[integration]} />
                    <Button
                      color="secondary"
                      className={classes.button}
                      onClick={() => this.addIntegration(integration)}
                      variant="contained"
                    >
                      + Add {integration}
                    </Button>
                  </div>
                ))
              }
            </div>
          </ExpansionPanelDetails>
        </ExpansionPanel>
      </div>
    )
  }
}

Integrations.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(Integrations);
