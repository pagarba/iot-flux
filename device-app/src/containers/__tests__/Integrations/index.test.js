import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { Integrations } from '../../Integrations';

describe('<Integrations />', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render components', () => {
    const wrapper = shallow(
      <Integrations
        classes={{}}
        location={{search: ''}}
        theme={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});