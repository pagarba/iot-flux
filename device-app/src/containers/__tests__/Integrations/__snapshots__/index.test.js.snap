// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Integrations /> should render components 1`] = `
<div>
  <WithStyles(ExpansionPanel)
    expanded={true}
    key="1"
    onChange={[Function]}
  >
    <WithStyles(ExpansionPanelSummary)
      expandIcon={
        <pure(ExpandMoreIcon)
          color="primary"
        />
      }
    >
      <WithStyles(Typography)
        color="primary"
      >
        MY INTEGRATIONS
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)>
      <div>
        <div
          key="0"
        >
          <img
            src="justifire.png"
          />
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            + Delete 
            justifire
          </WithStyles(Button)>
        </div>
        <div
          key="1"
        >
          <img
            src="visual_flights.png"
          />
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            + Delete 
            visualFlights
          </WithStyles(Button)>
        </div>
      </div>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
  <WithStyles(ExpansionPanel)
    expanded={false}
    key="2"
    onChange={[Function]}
  >
    <WithStyles(ExpansionPanelSummary)
      expandIcon={<pure(ExpandMoreIcon) />}
    >
      <WithStyles(Typography)>
        ADD A NEW INTEGRATION
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)>
      <div>
        <div>
          <img
            src="trigger_smart.png"
          />
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            + Add 
            triggerSmart
          </WithStyles(Button)>
        </div>
        <div>
          <img
            src="datalife.png"
          />
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            + Add 
            datalife
          </WithStyles(Button)>
        </div>
        <div>
          <img
            src="guniary.png"
          />
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            variant="contained"
          >
            + Add 
            guniary
          </WithStyles(Button)>
        </div>
      </div>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
</div>
`;
